cmd/mesg-daemon/main.go

```diff
@@ -35,12 +38,29 @@ func main() {
 	// init the config of cosmos
 	cosmos.InitConfig()
 
-	ctx := server.NewDefaultContext()
+	ctx := server.NewContext(
+		cfg.DefaultConfig(),
+		log.NewTMJSONLogger(log.NewSyncWriter(os.Stdout)),
+	)
 	cobra.EnableCommandSorting = false
 	rootCmd := &cobra.Command{
-		Use:               version.ServerName,
-		Short:             "Engine Daemon (server)",
-		PersistentPreRunE: server.PersistentPreRunEFn(ctx),
+		Use:   version.ServerName,
+		Short: "Engine Daemon (server)",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := server.PersistentPreRunEFn(ctx)(cmd, args); err != nil {
```

Comment:
I wraped the function `server.PersistentPreRunEFn` because it does some stuff using private function and it replaces the `context.Logger` inside the function (even if the logger is already initialized..)